File: Problems/1716.ts
describe("Typescript test:", () => {
    test("Testing", () => {
        expect(totalMoney(20)).toEqual(96);
    });
});

function totalMoney(n: number): number {
    let result = 0;
    let w = 0;
    let d = 1;
    for (let i = 1; i <= n; i++) {
        result += w + d;
        d++;
        if (!(i % 7)) {
            w++;
            d = 1;
        }
    }
    return result;
}

File: Problems/1716.rs
fn main() {}
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_case() {
        assert_eq!(total_money(10), 37);
    }
}

pub fn total_money(n: i32) -> i32 {
    let mut result = 0;
    let mut w = 0;
    let mut d = 0;
    for i in 1..=n {
        result += w + d + 1;
        d += 1;
        if i % 7 == 0 {
            w += 1;
            d = 0;
        }
    }
    result
}
